imageio/imageio-batik/src/main/java/com/twelvemonkeys/imageio/plugins/svg/SVGImageReaderSpi.java

```diff
@@ -64,97 +71,25 @@ public boolean canDecodeInput(final Object pSource) throws IOException {
         return pSource instanceof ImageInputStream && canDecode((ImageInputStream) pSource);
     }
 
-    @SuppressWarnings("StatementWithEmptyBody")
     private static boolean canDecode(final ImageInputStream pInput) throws IOException {
-        // NOTE: This test is quite quick as it does not involve any parsing,
-        // however it may not recognize all kinds of SVG documents.
+        QName doctype;
+        pInput.mark();
         try {
-            pInput.mark();
-
-            // TODO: This is not ok for UTF-16 and other wide encodings
-            // TODO: Use an XML (encoding) aware Reader instance instead
-            // Need to figure out pretty fast if this is XML or not
-            int b;
-            while (Character.isWhitespace((char) (b = pInput.read()))) {
-                // Skip over leading WS
-            }
-
-            // If it's not a tag, this can't be valid XML
-            if (b != '<') {
-                return false;
-            }
```

Comment:
What if we keep this, just to very quickly filter out any file that is *not* XML, before creating stream wrapper and attempting to parse?

We might have to strengthen the whitespace skipping to handle BOMs and other multibyte whitespace chars. But it would at least make me happier, knowing we won't spend time trying to XML-parse binary files... 😀

Reply:
The current detection is good enough, to be honest.  What it lacks most is likely handling UTF-16 and UTF-BOM.  Given nowadays almost everything uses just UTF-8 w/o BOM, one may consider my suggestion here superfluous. 😄  It just happens I've encountered this code and decided to give it a try.

I don't think you should spend more time looking into this PR.  Thank you for providing feedback to me also.

Reply:
Alternatively, the detection using an XML parser could be included and triggered by a system property, depending on one's needs.

Reply:
> ... over the weekend.

Over already.